proxy/proxy/client.go

```diff
@@ -219,6 +221,25 @@ func (c *Client) refreshCfg(instance string) (addr string, cfg *tls.Config, vers
 	certs := x509.NewCertPool()
 	certs.AddCert(scert)
 
+	certExpiration := mycert.Leaf.NotAfter
+	if c.Certs.IAMLoginEnabled() {
+		tokenExpiration, tokErr := c.Certs.TokenExpiration()
+		if tokErr != nil {
+			return "", nil, "", tokErr
+		}
+		if certExpiration.After(tokenExpiration) {
+			certExpiration = tokenExpiration
+		}
+	}
+	now := time.Now()
+	timeToRefresh := certExpiration.Sub(now) - refreshCfgBuffer
+	if timeToRefresh <= 0 {
+		err = fmt.Errorf("new ephemeral certificate expires too soon: current time: %v, certificate expires: %v", now, certExpiration)
+		logging.Errorf("ephemeral certificate (%+v) error: %v", mycert, err)
+		return "", nil, "", err
+	}
+	go c.refreshCertAfter(instance, timeToRefresh)
```

Comment:
Why move this up?

Reply:
Mostly as a convenience for reading, but I'll move it back down to make a smaller diff.